include/libp2p/injector/kademlia_injector.hpp

```diff
@@ -22,6 +22,45 @@
 
 namespace libp2p::injector {
 
+  template <typename Injector>
```

Comment:
This fn is currently used in Fuhon, to maintain shared_ptrs with limited lifetime